### ISSUE TITLE
unwrap panic from shared state

### ISSUE THREAD

**Issue by alexlatif** [geofmureithi/apalis#411]
stopping shared states being used across an axum application
```
pub trait FromData: Sized + Clone + Send + Sync + 'static {
    /// Gets the value
    fn get(data: &crate::data::Extensions) -> Self {
        data.get::<Self>().unwrap().clone()
    }
}
```
from this shared state
```
#[derive(Clone)]
pub struct AppState {
    config: AppConfig,
    redis_manager: RedisManager,
    orchestrator_manager: Arc<RwLock<dyn OrchestratorManager>>,
    service_register: Arc<RwLock<ServiceRegister>>,
}
```
passed into the worker 
```
    let py_worker_task = WorkerBuilder::new("py-worker")
        .layer(TraceLayer::new())
        .layer(LogLayer::new("py-worker"))
        .with_storage(py_queue.clone())
        .data(state.clone())
        .build_fn(publish_request_await_py_response);
```
defined as follows
```
pub async fn publish_request_await_py_response(
    job: PyWorkerJob,
    state: Data<AppState>,
    _worker_id: WorkerId,
) -> Result<(), JobError> {
```

**Comment by geofmureithi** [comment]
This looks really weird and unexpected. Could you try a few things:

- Have the `data` layer before the `with_storage` call.
- Ensure that state is of type `AppState`

I will try and replicate the issue.

Related to #393

**Comment by alexlatif** [comment]
yes confirmed with point no 2. AppState was wrapped in an Arc and the panic made it hard to debug. Perhaps a #[debug_worker] macro like in Axum could help with these details?

**Comment by geofmureithi** [comment]
Sure, this will be fixed by #393
